### DISCUSSION TITLE
argocd api create cluster sync app fail

### DISCUSSION THREAD

**Discussion by catsyuan** [argoproj/argo-cd#10121] [Q&A]
error：
Duration:           0s
Message:            ComparisonError: error getting openapi resources: unknown;ComparisonError: error getting openapi resources: unknown
FATA[0000] Operation has completed with phase: Error    

cluster 
status： Failed
DETAILS:  error getting openapi resources: unknown

I create cluster via api：/api/v1/clusters
and manual create  "argocd-manager" in namespace "kube-system" ，ClusterRole "argocd-manager-role"，ClusterRoleBinding "argocd-manager-role-binding" created

**Comment by jdmaguire** [comment]
Hi @catsyuan, I saw the same error message today. Did you ever find the issue for you?

**Comment by ZacharyU-1** [comment]
I have been seeing the same error on this as well. Was there ever any update?

I have been adding my Cluster to ArgoCD through the API which works. However, when I add the `Application` through the API, I am expecting it to deploy my manifest to my cluster. However, I am getting the same error as above : 
`ComparisonError: error getting openapi resources: unknown`
`ComparisonError: error getting openapi resources: unknown
FATA[0000] Operation has completed with phase: Error`

**Comment by jdmaguire** [comment]
My understanding of this issue is that this is somewhat a catch-all message.

For myself personally, I fixed the issue by creating a KSA and giving it the necessary permissions and ensuring that ArgoCD had the proper config to connect to the K8s cluster.

I was using Terraform at the time. [Here is a link](https://github.com/oboukili/terraform-provider-argocd/issues/194) to some discussion around this issue; again, keep in mind that this is with using a terraform provider. However you are installing ArgoCD will vary what is needed and it may be another issue entirely that you have.
